zio-http/shared/src/main/scala/zio/http/Response.scala

```diff
@@ -133,12 +133,14 @@ object Response {
 
   def badRequest(message: String): Response = error(Status.BadRequest, message)
 
-  def error(status: Status.Error, message: String): Response = {
+  def error(status: Status.Error, message: String, includeWarning: Boolean = false): Response = {
     import zio.http.internal.OutputEncoder
 
     val message2 = OutputEncoder.encodeHtml(if (message == null) status.text else message)
 
-    Response(status = status, headers = Headers(Header.Warning(199, "ZIO HTTP", message2)))
+    val headers = if (includeWarning) Headers(Header.Warning(199, "API", message2)) else Headers.empty
```

Comment:
I set the agent as `API` since `ZIO HTTP` can possibly be a security vulnerability, as mentioned on the issue description:
> Exposing backend implementation details (as I've said, this header mentions ZIO HTTP) can be considered a security issue.